subsys/net/openthread/rpc/client/ot_rpc_diag_client.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ot_rpc_ids.h>
+
+#include <nrf_rpc_cbor.h>
+
+#include <openthread/error.h>
+#include <openthread/dataset.h>
+#include <openthread/instance.h>
+#include <openthread/link.h>
+#include <openthread/thread.h>
+
+NRF_RPC_GROUP_DECLARE(ot_group);
+LOG_MODULE_DECLARE(ot_rpc, LOG_LEVEL_DBG);
+
+static char version[256];
+static otExtAddress mac = {.m8 = {0xF4, 0xCE, 0x36, 0x24, 0x95, 0x6A, 0xD1, 0xD0}};
+static otExtendedPanId ext_pan_id;
+static char thread_network_name[OT_NETWORK_NAME_MAX_SIZE + 1];
+static otMeshLocalPrefix mesh_prefix;
+
+static void get_uint_t(ot_rpc_id id, void *result, size_t result_size)
+{
+	const size_t cbor_buffer_size = 0;
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_cbor_cmd_rsp(&ot_group, id, &ctx);
+
+	if (!zcbor_uint_decode(ctx.zs, result, result_size)) {
+		nrf_rpc_err(-EBADMSG, NRF_RPC_ERR_SRC_RECV, &ot_group, id, NRF_RPC_PACKET_TYPE_CMD);
+	}
+
+	nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+}
+
+static int get_string(ot_rpc_id id, char *buffer, size_t buffer_size)
+{
+	const size_t cbor_buffer_size = 0;
+	size_t bytes_copied = 0;
+	struct zcbor_string zst;
+	struct nrf_rpc_cbor_ctx ctx;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+
+	nrf_rpc_cbor_cmd_rsp(&ot_group, id, &ctx);
+
+	if (!zcbor_bstr_decode(ctx.zs, &zst)) {
+		nrf_rpc_err(-EBADMSG, NRF_RPC_ERR_SRC_RECV, &ot_group, id, NRF_RPC_PACKET_TYPE_CMD);
+		goto exit;
+	}
+
+	bytes_copied = buffer_size < zst.len ? buffer_size : zst.len;
+	memcpy(buffer, zst.value, bytes_copied);
+
+exit:
+	nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+	return bytes_copied;
+}
+
+const char *otGetVersionString(void)
+{
+	get_string(OT_RPC_CMD_GET_VERSION_STRING, version, sizeof(version));
+
+	return version;
+}
+
+otError otDatasetGetActive(otInstance *aInstance, otOperationalDataset *aDataset)
+{
+	return 0;
+}
+
+uint8_t otLinkGetChannel(otInstance *aInstance)
+{
+	uint8_t ret;
+
+	get_uint_t(OT_RPC_CMD_LINK_GET_CHANNEL, &ret, sizeof(ret));
+
+	return ret;
+}
+
+const otMacCounters *otLinkGetCounters(otInstance *aInstance)
+{
+	return 0;
+}
+
+const otExtAddress *otLinkGetExtendedAddress(otInstance *aInstance)
+{
+	int ret_size = get_string(OT_RPC_CMD_LINK_GET_EXTENDED_ADDRESS, mac.m8, sizeof(mac.m8));
+
+	if (ret_size != sizeof(mac.m8)) {
+		LOG_ERR("Received mac addr size is too short");
```

Comment:
please check newer version.

Reply:
I guess this is possible to use in all such places this function: https://github.com/nrfconnect/sdk-nrf/blob/collab-serial-ble-ot/subsys/net/openthread/rpc/common/ot_rpc_common.h#L21
but it can be done as final clean up step that we discussed today